[Tooling] Report Firebase Test Failures as Buildkite Annotations

AliSoftware opened the pull request:
### What

This PR catches test failures happening when running Instrumented Tests in Firebase, and add a Buildkite annotation at the top of the failed build so it's way easier to jump to the details of the test failure on Firebase Console.

This is a reproduction of the similar changes that were done in https://github.com/wordpress-mobile/WordPress-Android/pull/17581

### How it looks

[This is how the annotation appears on Buildkite](https://buildkite.com/automattic/woocommerce-android/builds/8193) if an Instrumented Test failure occurs:

<img width="1168" alt="image" src="https://user-images.githubusercontent.com/216089/205111947-8929410d-4083-472e-9511-508d6b72a4ce.png">

### To Test

Observe that the intentional failure that I've forced to happen in 2065505ea7d7677b37c0d8ff35d251210140de22 indeed [generated a Buildkite annotation](https://buildkite.com/automattic/woocommerce-android/builds/8193) (see screenshot above) — while a [successful Instrumented Tests run like the one at the HEAD of this branch](https://buildkite.com/automattic/woocommerce-android/builds/latest?branch=tooling/firebase-test-failure-annotation) doesn't.

AliSoftware committed:
Annotate Firebase Test failures on Buildkite

AliSoftware committed:
Point to latest `release-toolkit`

AliSoftware committed:
🚧 Introduce fake failure in Instrumented Tests

To test FTL failure reporting

AliSoftware committed:
Revert "🚧 Introduce fake failure in Instrumented Tests"

This reverts commit 2065505ea7d7677b37c0d8ff35d251210140de22.

mokagio commented on fastlane/Fastfile:
```diff
@@ -734,16 +734,27 @@ platform :android do
+      crash_on_test_failure: false
      )
+
+     annotation_ctx = 'firebase-test-woocommerce-vanilla-debug'
+     if test_succeeded
+      sh("buildkite-agent annotation remove --context '#{annotation_ctx}' || true") if is_ci?
+     else
+      details_url = lane_context[SharedValues::FIREBASE_TEST_MORE_DETAILS_URL]
+      message = "Firebase Tests failed. Failure details can be seen [here in Firebase Console](#{details_url})"
+      sh('buildkite-agent', 'annotate', message, '--style', 'error', '--context', annotation_ctx) if is_ci?
+      UI.test_failure!(message)
+    end
```
🔍 🔍 🔍  the indentation here is one space instead of two. I assume was an editor blunder rather than something intentional?

```suggestion
     annotation_ctx = 'firebase-test-woocommerce-vanilla-debug'
     if test_succeeded
       sh("buildkite-agent annotation remove --context '#{annotation_ctx}' || true") if is_ci?
     else
       details_url = lane_context[SharedValues::FIREBASE_TEST_MORE_DETAILS_URL]
       message = "Firebase Tests failed. Failure details can be seen [here in Firebase Console](#{details_url})"
       sh('buildkite-agent', 'annotate', message, '--style', 'error', '--context', annotation_ctx) if is_ci?
       UI.test_failure!(message)
    end
```

AliSoftware replied:
Nice catch.

Strangely, we had the exact same indentation issue in WPAndroid, but I forgot to fix it here in Woo.

Fixed in 8fdd588f5eafd0212ae67adc5df286196da394f4 (a little bit differently than your code suggestion above, because the indentation issue was actually the other way around 😄 )

mokagio commented on fastlane/Fastfile:
```diff
@@ -734,16 +734,27 @@ platform :android do
+      crash_on_test_failure: false
      )
+
+     annotation_ctx = 'firebase-test-woocommerce-vanilla-debug'
```
Nice that this is `woocommerce`, not `wordpress`. It would have been easy to miss when porting the code.

👨‍🍳 💋

mokagio reviewed (approved):
I disabled auto-merge because of the indentation nitpick

AliSoftware committed:
Fix indentation

Kudos @mokagio for catching it in https://github.com/woocommerce/woocommerce-android/pull/7976#discussion_r1037362011
